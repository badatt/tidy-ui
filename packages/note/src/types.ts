import { HTMLAttributes } from 'react';
import { IBaseProps } from '@tidy-ui/types';

/**
 * NoteProps
 */
export interface INoteProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {}
