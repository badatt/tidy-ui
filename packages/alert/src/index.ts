export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
