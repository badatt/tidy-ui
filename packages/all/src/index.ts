export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
