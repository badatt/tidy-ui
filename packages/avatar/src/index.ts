export * from './Avatar';
export * from './AvatarBadge';
export * from './AvatarGroup';
