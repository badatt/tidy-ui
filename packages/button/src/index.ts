export * from './Button';
export * from './Anchor';
