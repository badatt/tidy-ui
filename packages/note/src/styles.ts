import { color, css, hsla } from '@tidy-ui/commons';
import { INoteProps } from './types';

const baseStyles = css<INoteProps>`
  ${({ theme: { isDark } }) => css`
    border-left: 4px solid ${isDark ? hsla(color.yellow[800]) : hsla(color.yellow[500])};
    background-color: ${isDark ? hsla(color.yellow[700], 0.2) : hsla(color.yellow[400], 0.2)};
  `}
`;

const realStyles = css<INoteProps>`
  ${({ theme: { isDark } }) => css`
    ${isDark
      ? css`
          box-shadow: rgb(164 164 200 / 25%) 0px 13px 27px -5px, rgb(147 147 147 / 15%) 0px 8px 16px -8px;
        `
      : css`
          box-shadow: rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;
        `}
  `}
`;

export { baseStyles, realStyles };
