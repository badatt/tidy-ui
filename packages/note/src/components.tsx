import { applyStandardOverrideStyles, createFontStyle, css, styled } from '@tidy-ui/commons';
import { baseStyles, realStyles } from './styles';
import { INoteProps } from './types';

const NoteRoot = styled.div<INoteProps>`
  position: relative;
  ${createFontStyle()}
  padding: 0.75rem 1rem;
  ${({ isReal }) => css`
    ${isReal ? realStyles : baseStyles}
  `}
  ${applyStandardOverrideStyles}
`;

export { NoteRoot };
