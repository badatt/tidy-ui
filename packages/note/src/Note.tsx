import React from 'react';
import { NoteRoot } from './components';
import { INoteProps } from './types';

const Note = React.forwardRef<HTMLDivElement, INoteProps>((props, ref) => {
  const { children, ele, ...rest } = props;
  return (
    <NoteRoot ref={ref} role="note" {...rest}>
      {ele ? React.cloneElement(ele, {}, children) : children}
    </NoteRoot>
  );
});

Note.defaultProps = {};

Note.displayName = 'Note';

export { Note };
