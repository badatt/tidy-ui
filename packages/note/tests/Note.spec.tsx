/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight, TidyUiProvider } from '../../commons/src';
import { Note } from '../src';

describe('Note', () => {
  it('Basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Note>basic</Note>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Dark mode basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <Note>basic</Note>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Reality render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Note isReal>reality</Note>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Dark mode reality render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <Note isReal>reality</Note>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Custom element', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Note ele={<div />}>basic</Note>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
