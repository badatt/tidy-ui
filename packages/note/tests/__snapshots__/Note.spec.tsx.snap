// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  border-left: 4px solid hsla(45,93%,47%,1);
  background-color: hsla(48,96%,53%,0.2);
}

<body>
    <div>
      <div
        class="c0"
        role="note"
      >
        basic
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  border-left: 4px solid hsla(45,93%,47%,1);
  background-color: hsla(48,96%,53%,0.2);
}

<div>
    <div
      class="c0"
      role="note"
    >
      basic
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Note Custom element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  border-left: 4px solid hsla(45,93%,47%,1);
  background-color: hsla(48,96%,53%,0.2);
}

<body>
    <div>
      <div
        class="c0"
        role="note"
      >
        <div>
          basic
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  border-left: 4px solid hsla(45,93%,47%,1);
  background-color: hsla(48,96%,53%,0.2);
}

<div>
    <div
      class="c0"
      role="note"
    >
      <div>
        basic
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Note Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  border-left: 4px solid hsla(45,93%,47%,1);
  background-color: hsla(48,96%,53%,0.2);
}

<body>
    <div>
      <div
        class="c0"
        role="note"
      >
        basic
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  border-left: 4px solid hsla(45,93%,47%,1);
  background-color: hsla(48,96%,53%,0.2);
}

<div>
    <div
      class="c0"
      role="note"
    >
      basic
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Note Dark mode reality render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  box-shadow: rgba(50,50,93,0.25) 0px 13px 27px -5px,rgba(0,0,0,0.3) 0px 8px 16px -8px;
}

<body>
    <div>
      <div
        class="c0"
        role="note"
      >
        reality
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  box-shadow: rgba(50,50,93,0.25) 0px 13px 27px -5px,rgba(0,0,0,0.3) 0px 8px 16px -8px;
}

<div>
    <div
      class="c0"
      role="note"
    >
      reality
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Note Reality render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  box-shadow: rgba(50,50,93,0.25) 0px 13px 27px -5px,rgba(0,0,0,0.3) 0px 8px 16px -8px;
}

<body>
    <div>
      <div
        class="c0"
        role="note"
      >
        reality
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
  padding: 0.75rem 1rem;
  box-shadow: rgba(50,50,93,0.25) 0px 13px 27px -5px,rgba(0,0,0,0.3) 0px 8px 16px -8px;
}

<div>
    <div
      class="c0"
      role="note"
    >
      reality
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
